File: Number of Lines to Write String (806)/806.py
#
#

class Solution:
    def numberOfLines(self, widths, S):
        """
        :type widths: List[int]
        :type S: str
        :rtype: List[int]
        """
        total_width = 0
        total_lines = 1
    
        for letter in S:
            curr_width = widths[ord(letter) - 97]
            total_width += curr_width
            if total_width > 100:
                total_lines += 1
                total_width = curr_width

        return [total_lines, total_width]

File: Number of Lines to Write String (806)/806.java
// Runtime: 2 ms
// Beats 88.86% of Java solutions

class Solution {
    public int[] numberOfLines(int[] widths, String S) {
        int totalWidth = 0;
        int totalLines = 1;
        
        for (char letter : S.toCharArray()) {
            int currWidth = widths[letter - 'a'];
            totalWidth += currWidth;
            
            if (totalWidth > 100) {
                totalLines++;
                totalWidth = currWidth;
            }
        }
        
        return new int[] {totalLines, totalWidth};
    }
}